girder/utility/filesystem_assetstore_adapter.py

```diff
@@ -106,6 +107,7 @@ def __init__(self, assetstore):
         # happens to existing assetstores that no longer can access their temp
         # directories.
         self.tempDir = os.path.join(self.assetstore['root'], 'temp')
+        self.deleteLock = threading.Lock()
```

Comment:
We can use a file lock, instead.  This is heavier, of course. 

Do we have deployed instances with multiple app servers?  Do we test for other resource contention issues?